packages/components/table/src/table/defaults.ts

```diff
@@ -179,7 +180,10 @@ export default {
     type: Array as PropType<DefaultRow[]>,
     default: () => [],
   },
-  size: String,
+  size: {
+    type: String as PropType<Size>,
+    default: 'default',
+  },
```

Comment:
```suggestion
  size: useSizeProp,
```

refer [Button.ts](https://github.com/element-plus/element-plus/blob/dev/packages/components/button/src/button.ts#L26)